### PR TITLE
feat: scaleUpCooldownSeconds and scaleDownCooldownSeconds to replace cooldownSeconds

### PR DESCRIPTION
**Pull request by whynowy**
A way to control different speed for scaling up and down.

### CODE REVIEW

File: pkg/apis/numaflow/v1alpha1/vertex_types.go [hunk 1]
```diff
@@ -580,6 +581,14 @@ type Scale struct {
 	// The is use to prevent too aggressive scaling operations
 	// +optional
 	ReplicasPerScale *uint32 `json:"replicasPerScale,omitempty" protobuf:"varint,9,opt,name=replicasPerScale"`
+	// ScaleUpCooldownSeconds defines the cooldown seconds after a scaling operation, before a follow-up scaling up.
+	// It defaults to the CooldownSeconds if not set.
+	// +optional
+	ScaleUpCooldownSeconds *uint32 `json:"scaleUpCooldownSeconds,omitempty" protobuf:"varint,11,opt,name=scaleUpCooldownSeconds"`
```
Comment (KeranYang) [inline pkg/apis/numaflow/v1alpha1/vertex_types.go:587]: Do we skip the number 10 on purpose and why?

**Review by KeranYang — APPROVED** [review]

File: pkg/apis/numaflow/v1alpha1/vertex_types.go (hunk 1 above)
Comment (whynowy) [reply pkg/apis/numaflow/v1alpha1/vertex_types.go:587]: Ooops, Copilot generated.....

File: pkg/reconciler/vertex/scaling/scaling.go [hunk 2]
```diff
@@ -309,27 +312,35 @@ func (s *Scaler) scaleOneVertex(ctx context.Context, key string, worker int) err
 		if diff > maxAllowed {
 			diff = maxAllowed
 		}
+		if secondsSinceLastScale < scaleDownCooldown {
```
Comment (vigith) [inline pkg/reconciler/vertex/scaling/scaling.go:315]: move it after 176?

Comment (whynowy) [reply pkg/reconciler/vertex/scaling/scaling.go:315]: why?

Comment (vigith) [reply pkg/reconciler/vertex/scaling/scaling.go:315]: aren't we doing the check there, and this equality for change?

Comment (whynowy) [reply pkg/reconciler/vertex/scaling/scaling.go:315]: This check happens after we know it needs to scale down.

**Review by vigith — APPROVED** [review]

**Pull request merged**